aten/src/ATen/native/native_functions.yaml

```diff
@@ -10514,6 +10513,16 @@
 - func: linalg_multi_dot.out(Tensor[] tensors, *, Tensor(a!) out) -> Tensor(a!)
   python_module: linalg
 
+- func: linalg_tensordot(Tensor self, Tensor other, int[] dims_self, int[] dims_other) -> Tensor
```

Comment:
The signature in Python is `tensordot(a, b, dims=2, out=None)` where `dims (int or Tuple[List[int], List[int]] or List[List[int]] containing two lists or Tensor)`. Can all the logic on `dims` argument be ported from Python to C++? If not maybe the Python signature should be modified so that the mapping from Python to C++ is more direct and no glue code from functional.py is neccessary?

Reply:
honestly, I don't know the answer to this question. The fact that the logic was originally written in Python makes me to suspect that there was a good reason to do that but I don't really know if the original reason still holds.
E.g., maybe the dispatcher cannot handle such a complex overload?